recipes-bsp/u-boot/u-boot-distro-boot/uEnv.txt.in

```diff
@@ -18,7 +22,7 @@ set_kernel_load_addr=env set bootcmd_unzip_k ';'; \
                              env set kernel_addr_load ${loadaddr}; \
                          elif test ${kernel_image_type} = "Image.gz"; then \
                              env set kernel_addr_load ${loadaddr}; \
-                             env set bootcmd_unzip_k 'unzip $kernel_addr_load $kernel_addr_r'; \
+                             env set bootcmd_unzip_k "unzip $kernel_addr_load $kernel_addr_r"; \
```

Comment:
Yes this was intended, the single quotes here was causing this particular variable to not be translated correctly during `set_vars`, due to the back-and-forth from shell variable to environment variable again. I couldn't find a way to have the original single quotes be maintained during this transformation. I also rather not have an exception in the logic just for this one variable since nothing else in `uEnv.txt` uses single quotes except this.

I also made sure to check and while this does cause `kernel_addr_load` and `kernel_addr_load_r` to be evaluated right away. The value of both of these variables do not get changed/modified at all between when `set_kernel_load_addr` is run and when `bootcmd_unzip_k` is run. Meaning I didn't see any actual technical reasons for this line to use single quotes in the first place. Since the end result of evaluating these variables ends up being the same whether single or double quotes get used here.

In the end I found it easier to just change the quotations used here then to try and figure out how to handle single quotes in `set_vars`. I would consider otherwise if other places in `uEnv.txt` were also using single quotes but this seems to be the only place it's used. And as I said from a technical point of view it doesn't really seem like it needed to use single quotes anyways.

Reply:
Thanks for the confirmation. That seems a perfectly fine approach.